lib/main.js

```diff
@@ -54,7 +54,9 @@ module.exports = {
 
     this.subscriptions.add(atom.workspace.observeTextEditors(editor => {
       editor.onDidSave(() => {
-        if (scopes.indexOf(editor.getGrammar().scopeName) !== -1 && atom.config.get('linter-eslint.fixOnSave')) {
+        const editorScopes = editor.getLastCursor().getScopeDescriptor().scopes;
+        const validScope = editorScopes.some(scope => scopes.includes(scope));
```

Comment:
I don't know the performance impact of checking the scope, but we could probably first check that `fixOnSave` is true before calculating that the scope is valid, right?